datashader/glyphs.py

```diff
@@ -29,10 +29,12 @@ def validate(self, in_dshape):
             raise ValueError('y must be real')
 
     def _compute_x_bounds(self, df):
-        return df[self.x].min(), df[self.x].max()
+        xs = df[self.x].values
+        return xs.min(), xs.max()
```

Comment:
@jbednar I had some difficulties using numba with dask. For this reason there are now two versions of the bounds computations: one is for pandas dataframes, relying on numba and only doing one pass through the data. The other is a memoized version of the existing code (except with `np.nanmin` and `np.nanmax`). The memoization gets us the same performance as cachey did - which is great news - but still does two passes through the data. The memoization does not work with pandas dataframes because they're mutable, whereas dask dataframes are not. However there is still a solid speedup to the pandas side of datashader.